### PR TITLE
implemented changes

### PR DESCRIPTION
**Pull request by vshand11**
Fix tsv file with empty header issue

### CODE REVIEW

File: src/essential_validator.py [hunk 1]
```diff
@@ -262,8 +269,9 @@ def validate_data(self, file_info):
             return False
         
         # Each row in a metadata file must have same number of columns as the header row
-        if '' in self.df.columns:
-            msg = f'Invalid metadata, headers are match row columns, {self.batch[ID]}!'
+        empty_cols = [col for col in self.df.columns.tolist() if "Unnamed:" in col or not col]
```
Comment (knockknockyoo) [inline src/essential_validator.py:272]: Where is "Unnamed" string coming from? Can we avoid this? If not, we should create a constant variable. And, we should leave a comment.

File: src/essential_validator.py [hunk 2]
```diff
@@ -143,19 +143,26 @@ def validate(self,batch):
...
-            #2. validate meatadata in self.df
-            if not self.validate_data(file_info):
-                file_info[STATUS] = "failed"
-                return False
-        
+        try:
+            for file_info in self.file_info_list: 
+                #1. download the file in s3 and load tsv file into dataframe
+                if not self.download_file(file_info):
+                    file_info[STATUS] = "failed"
```
Comment (knockknockyoo) [inline src/essential_validator.py:150]: Can we create a constant for "failed"?

File: src/essential_validator.py [hunk 3]
```diff
@@ -143,19 +143,26 @@ def validate(self,batch):
...
+                    return False
+                #2. validate meatadata in self.df
+                if not self.validate_data(file_info):
+                    file_info[STATUS] = "failed"
```
Comment (knockknockyoo) [inline src/essential_validator.py:154]: Can we create a constant for "failed"?

File: src/essential_validator.py [hunk 4]
```diff
@@ -95,7 +95,7 @@ def essentialValidate(configs, job_queue, mongo_dao):
                             mongo_dao.set_submission_validation_status(validator.submission, None, submission_meta_status, None)
                     else:
                         log.error(f'Invalid message: {data}!')
-
+                    log.info(f'Processed the job!')
```
Comment (knockknockyoo) [inline src/essential_validator.py:98]: Ideally, we can change the log like "Processed the job; metadata/file/essential"

File: src/data_loader.py [hunk 5]
```diff
@@ -114,6 +117,9 @@ def load_data(self, file_path_list):
                     self.log.exception(msg)
                     self.errors.append(msg)
                     return False, self.errors
+            finally:
+                deleted_nodes = None
+                deleted_file_nodes = None
```
Comment (knockknockyoo) [inline src/data_loader.py:122]: After this "load_data" function, deleted_nodes, deleted_file_nodes will be de-allocated from the heap memory. I think we dont need to set None. Could you double check?

**Review by knockknockyoo — CHANGES_REQUESTED** [review]

File: src/essential_validator.py (hunk 1 above)
Comment (vshand11) [reply src/essential_validator.py:272]: during dataframe parsing the tsv file.

File: src/data_loader.py (hunk 5 above)
Comment (vshand11) [reply src/data_loader.py:122]: explicitly set to None to make memory allocated immediately.

**Review by knockknockyoo — APPROVED** [review]

**Pull request merged**